drivers/scheduler/eks/eks.go

```diff
@@ -0,0 +1,304 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/sched-ops/task"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	kube "github.com/portworx/torpedo/drivers/scheduler/k8s"
+	"github.com/portworx/torpedo/pkg/log"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	// SchedName is the name of the kubernetes scheduler driver implementation
+	SchedName = "eks"
+	// defaultEKSUpgradeTimeout is the default timeout for EKS control plane upgrade
+	defaultEKSUpgradeTimeout = 90 * time.Minute
+	// defaultEKSUpgradeRetryInterval is the default retry interval for EKS control plane upgrade
+	defaultEKSUpgradeRetryInterval = 5 * time.Minute
+)
+
+type EKS struct {
+	kube.K8s
+	clusterName     string
+	region          string
+	config          aws.Config
+	eksClient       *eks.Client
+	pxNodeGroupName string
+}
+
+// String returns the string name of this driver.
+func (e *EKS) String() string {
+	return SchedName
+}
+
+func (e *EKS) Init(schedOpts scheduler.InitOptions) (err error) {
+	// This implementation assumes the EKS cluster has two node groups: one group for
+	// Torpedo and another group for Portworx.
+	err = e.K8s.Init(schedOpts)
+	if err != nil {
+		return err
+	}
+	torpedoNodeName := ""
+	pods, err := core.Instance().GetPods("default", nil)
+	if err != nil {
+		log.Errorf("failed to get pods from default namespace. Err: [%v]", err)
+	}
+	if pods != nil {
+		for _, pod := range pods.Items {
+			if pod.Name == "torpedo" {
+				torpedoNodeName = pod.Spec.NodeName
+			}
+		}
+	}
+	nodes, err := core.Instance().GetNodes()
+	if err != nil {
+		log.Errorf("failed to get nodes. Err: [%v]", err)
+	}
+	e.region = os.Getenv("EKS_CLUSTER_REGION")
+	if e.region == "" {
+		nodeRegionLabel := "topology.kubernetes.io/region"
+		log.Warnf("env EKS_CLUSTER_REGION not found. Using node label [%s] to determine region", nodeRegionLabel)
+		if torpedoNodeName != "" && nodes != nil  {
+			for _, node := range nodes.Items {
+				if node.Name != torpedoNodeName {
+					e.region = node.Labels[nodeRegionLabel]
+					log.Infof("Using node label [%s] to determine region [%s]", nodeRegionLabel, e.region)
+					break
+				}
+			}
+		}
+		if e.region == "" {
+			return fmt.Errorf("env EKS_CLUSTER_REGION or node label [%s] not found", nodeRegionLabel)
+		}
+	}
+	e.pxNodeGroupName = os.Getenv("EKS_PX_NODEGROUP_NAME")
```

Comment:
Same here, do not add this as part of `init()`, move it to `UpgradeScheduler()`, as this is the place where its needed

Reply:
Done.